### ISSUE TITLE
when use "--weakly", all the IoU turned to 0

### ISSUE THREAD

**Issue by gxy1208** [fcdl94/WILSON#14]
hi! thanks for sharing the code.
I have met some problems:
I ran step0 , which is quite normal but when turned to step1 all the IoU came to 0 
![{PZS6XIGI5R{9%F~)EE724H](https://github.com/fcdl94/WILSON/assets/153416650/a81ae099-f41e-4ea2-9821-3627c0fc07ee)

the difference between two commands was using "--weakly"
is there any changes for the code?
thanks!

**Comment by fcdl94** [comment]
Hey! Can you please provide the full commands to better understand the issue?

**Comment by gxy1208** [comment]
exp1：
`exp --name Base --step 0 --lr 0.01 --bce --dataset voc --task 15-5 --batch_size 24 --epochs 30 --val_interval 2`
exp2：
```
pretr_FT=${path}Base_0.pth
exp --name OURS --step 1 --weakly --alpha 0.5 --lr 0.01 --step_ckpt $pretr_FT \
--loss_de 1 --lr_policy warmup --affinity --dataset voc --task 15-5 --batch_size 24 --epochs 20
```
There is not much difference between this and your code，so I'm not sure whether other points need to be modified

**Comment by fcdl94** [comment]
There is a mistake in the `lr` for the second run. 
As you can see from https://github.com/fcdl94/WILSON/blob/main/run.sh, the learning rate for the steps s > 0 should be `0.001`. It may fix your issue, which is probably some divergence.

Please, update me if this fixes the issue.

**Comment by gxy1208** [comment]
it still doesn't work
```
dataset=voc
epochs=20
task=15-5
lr_init=0.01
lr=0.001

dataset_pars="--dataset ${dataset} --task ${task} --batch_size 24 $ov --val_interval 2"

exp --name Base --step 0 --bce --lr ${lr_init} ${dataset_pars}  --epochs 30

pretr_FT=${path}Base_0.pth

exp --name Our --step 1 ${dataset_pars} --alpha 0.5 --lr ${lr} --step_ckpt $pretr_FT \
 --loss_de 1 --lr_policy warmup --affinity --epochs ${epochs}
```

**Comment by fcdl94** [comment]
Well, I don't know what the issue may be. 
Are you getting good performances on the first step? Have you tried running it without --weakly?

**Comment by gxy1208** [comment]
yes, good performances on the first step.
running without "weakly" also goes well

**Comment by fcdl94** [comment]
Can you please check that the version of torch and cuda are the same?

**Comment by gxy1208** [comment]
torch 1.13.1 cuda 11.6

**Comment by fcdl94** [comment]
Hey. Any news on the issue? I checked the version I was using: pytorch 1.6 with CUDA 10.1. 
If the issue is not fixed can you try with that?

**Comment by fcdl94** [comment]
Otherwise, in #13 it has showed that  with cuda12, pytorch2.1, GPU rtx4090, performance has been replicated.

**Comment by gxy1208** [comment]
hey. I have solved that issue! thank u so much~

**Comment by gxy1208** [comment]
hi! I wonder know how these files generate? if I change another dataset and how to prepare these files?
<img width="431" alt="image" src="https://github.com/fcdl94/WILSON/assets/153416650/61aca955-1ff5-4ab7-b75e-5791621a83e2">
<img width="417" alt="image" src="https://github.com/fcdl94/WILSON/assets/153416650/f4c588ae-926f-42e0-a121-a4f8a5d88c1c">

**Comment by fcdl94** [comment]
Hey @gxy1208.

I basically made a one_hot encoding for the classes in the image.
You can do a for loop on the segmentation annotations, make something like cls = np.unique(annotations) and then do a one_hot encoding that is 1 if index is in cls, 0 otherwise.

Hope it helps.
